azurecompute-arm/src/main/java/org/jclouds/azurecomputearm/domain/ResourceGroup.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.azurecomputearm.domain;
+
+import com.google.auto.value.AutoValue;
+import java.util.HashMap;
+import org.jclouds.javax.annotation.Nullable;
+import org.jclouds.json.SerializedNames;
+
+
+@AutoValue
+public abstract class ResourceGroup {
+
+   @AutoValue
+   public abstract static class ResourceGroupProperties{
+
+      public ResourceGroupProperties() {
```

Comment:
When using Google Auto-value, constructors shouldn't be used. A best practice is to declare them package-private instead of public to enforce that.